unity-renderer/Assets/Rendering/LoadingAvatar/CrossSection/BaseAvatarReveal.cs

```diff
@@ -121,13 +96,37 @@ public async UniTask StartAvatarRevealAnimation(bool withTransition, Cancellatio
             isRevealing = true;
             animation.Play();
             await UniTask.WaitUntil(() => !isRevealing, cancellationToken: cancellationToken).AttachExternalCancellation(cancellationToken);
+            StopCoroutine(updateAvatarRevealerRoutine);
         }
         catch(OperationCanceledException)
         {
             SetFullRendered();
         }
     }
 
+    private IEnumerator UpdateRevealerPosition()
+    {
+        _ghostMaterial.SetVector("_RevealPosition", revealer.transform.localPosition);
+        foreach (Material m in _materials)
+        {
+            m.SetVector("_RevealPosition", -revealer.transform.localPosition);
+        }
+        yield return new WaitForSeconds(0.01f);
+        updateAvatarRevealerRoutine = StartCoroutine(UpdateRevealerPosition());
```

Comment:
Wouldn't be much clearer to just do a `while(true)` with a `yield return null;` at the end of it?

Reply:
agree